src/app/components/matrix/matrix.component.ts

```diff
@@ -1,62 +1,94 @@
-import { AsyncPipe, NgFor } from '@angular/common';
-import { Component, OnInit } from '@angular/core';
-import { UntilDestroy, untilDestroyed } from '@ngneat/until-destroy';
 import { GameState } from '@angular-tetris/interface/game-state';
 import { Tile } from '@angular-tetris/interface/tile/tile';
 import { MatrixUtil } from '@angular-tetris/interface/utils/matrix';
-import { TetrisQuery } from '@angular-tetris/state/tetris/tetris.query';
-import { combineLatest, Observable, of, timer } from 'rxjs';
-import { map, switchMap, takeWhile } from 'rxjs/operators';
+import { TetrisStateService } from '@angular-tetris/state/tetris/tetris.state';
+import { NgFor } from '@angular/common';
+import {
+  ChangeDetectionStrategy,
+  Component,
+  DestroyRef,
+  effect,
+  inject,
+  signal
+} from '@angular/core';
 import { TileComponent } from '../tile/tile.component';
-@UntilDestroy()
 @Component({
   selector: 't-matrix',
   standalone: true,
-  imports: [TileComponent, NgFor, AsyncPipe],
+  imports: [TileComponent, NgFor],
   templateUrl: './matrix.component.html',
-  styleUrls: ['./matrix.component.scss']
+  styleUrls: ['./matrix.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush
 })
-export class MatrixComponent implements OnInit {
```

Comment:
Thanks so much for the effort. However, I have the same comment for the `LogoComponent`. 
Let's stick with the old RxJS implement here in `MatrixComponent ` as well.